cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/FileFetcherHttpHandlerInternalTest.java

```diff
@@ -0,0 +1,162 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.util.Random;
+
+/**
+ * Test for {@link FileFetcherHttpHandlerInternal}.
+ */
+public class FileFetcherHttpHandlerInternalTest {
+  @ClassRule
+  public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
+  private static final Logger LOG = LoggerFactory.getLogger(FileFetcherHttpHandlerInternalTest.class);
+  private static NettyHttpService httpService;
+  private static URL baseURL;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    CConfiguration cConf = CConfiguration.create();
+    cConf.set(Constants.CFG_LOCAL_DATA_DIR, TEMP_FOLDER.newFolder().getAbsolutePath());
+
+    Injector injector = Guice.createInjector(
+      new ConfigModule(cConf),
+      new LocalLocationModule());
+
+    FileFetcherHttpHandlerInternal handler = injector.getInstance(FileFetcherHttpHandlerInternal.class);
+
+    httpService = NettyHttpService.builder("FileFetcherHttpHandlerInternalTest")
+      .setHttpHandlers(handler)
+      .setExceptionHandler(new HttpExceptionHandler())
+      .build();
+
+    httpService.start();
+
+    InetSocketAddress addr = httpService.getBindAddress();
+    baseURL = new URL(String.format("http://%s:%d", addr.getHostName(), addr.getPort()));
+  }
+
+  @AfterClass
+  public static void stop() throws Exception {
+    httpService.stop();
+  }
+
+  @Test
+  public void testSuccess() throws Exception {
+    // Create a file.
+    File srcFile = TEMP_FOLDER.newFile("src_file");
+    try (FileOutputStream out = new FileOutputStream(srcFile)) {
+      byte[] bytes = new byte[256 * 1024 + 7];
+      new Random().nextBytes(bytes);
+      out.write(bytes);
+    }
+
+    // Download the file.
+    File targetFile = new File(TEMP_FOLDER.newFolder(), "dst_file");
+    Location dst = new LocalLocationFactory().create(targetFile.toURI());
+    HttpResponse httpResponse = download(srcFile, dst);
+    Assert.assertEquals(httpResponse.getResponseCode(), HttpResponseStatus.OK.code());
+
+    // Verify the source and destination files are identical.
+    byte[] orgFileContent = Files.readAllBytes(srcFile.toPath());
+    byte[] downloadedFileContent = Files.readAllBytes(targetFile.toPath());
+    Assert.assertArrayEquals(orgFileContent, downloadedFileContent);
+  }
+
+  @Test
+  public void testFileNotFound() throws Exception {
+    // A non-existent file
+    File srcFile = new File(TEMP_FOLDER.getRoot(), "file_dont_exist");
+
+    // Download the file.
+    File targetFile = new File(TEMP_FOLDER.newFolder(), "dst_file");
+    Location dst = new LocalLocationFactory().create(targetFile.toURI());
+
+    HttpResponse httpResponse = download(srcFile, dst);
+    Assert.assertEquals(httpResponse.getResponseCode(), HttpResponseStatus.NOT_FOUND.code());
```

Comment:
The expected value should be the first parameter

Reply:
Done.